gossip/bloom.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"encoding/binary"
+	"hash"
+	"time"
+
+	bloomfilter "github.com/holiman/bloomfilter/v2"
+	"golang.org/x/exp/rand"
+)
+
+var _ hash.Hash64 = (*hasher)(nil)
+
+func NewBloomFilter(m uint64, p float64) (*BloomFilter, error) {
+	bloom, err := bloomfilter.NewOptimal(m, p)
+	if err != nil {
+		return nil, err
+	}
+
+	bloomFilter := &BloomFilter{
+		Bloom: bloom,
+		Salt:  randomSalt(),
+	}
+	return bloomFilter, nil
+}
+
+type BloomFilter struct {
+	Bloom *bloomfilter.Filter
+	Salt  []byte
```

Comment:
cc: @StephenButtolph on how the salted bloom filter is being implemented as this will be used for the peer-list gossip implementation as well